kornia/filters/core_median.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+import torch.nn.functional as F
+
+from kornia.core import Module, IntegratedTensor
+from kornia.core.check import KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
+
+from .kernels import _unpack_2d_ks, get_binary_kernel2d
+
+import keras_core as keras
+
+
+def _compute_zero_padding(kernel_size: tuple[int, int] | int) -> tuple[int, int]:
+    r"""Utility function that computes zero padding tuple."""
+    ky, kx = _unpack_2d_ks(kernel_size)
+    return (ky - 1) // 2, (kx - 1) // 2
+
+def _median(input: IntegratedTensor, axis: int):
+    r"""Compute the median along the given dimension."""
+    
+    x = keras.ops.reshape(input, [-1, input.shape[axis]])
+    x = keras.ops.sort(x, axis=0)
+    middle_index = input.shape[axis] // 2
+
+    if input.shape[axis] % 2 != 0:
+        median_value, _ = keras.ops.top_k(x, k=middle_index, sorted=True)
+    
+    else:
+        median_value_1, _ = keras.ops.top_k(x, k=middle_index, sorted=True)
+        median_value_2, _ = keras.ops.top_k(x, k=middle_index, sorted=True)
+        median_value = median_value_1 + median_value_2 // 2
+
+    median_value = median_value[:, 0]
+
+    new_shape = input.shape.as_list()
+    new_shape[axis] = 1
+    tensor = keras.ops.reshape(median_value, new_shape)
+    
+    return tensor
+
+
+
+def median_blur(input: IntegratedTensor, kernel_size: tuple[int, int] | int) -> IntegratedTensor:
```

Comment:
wondering if we should have a decorator to auto cast to IntegratedTensor here. For the high level api, i'd like to explore a similar pattern e.g that opencv follows with the Input/Output Array api.

Reply:
+1